ocs_ci/deployment/flexy.py

```diff
@@ -87,8 +87,8 @@ def deploy_prereq(self):
         if not config.ENV_DATA.get('flexy_env_file'):
             self.clone_and_unlock_ocs_private_conf()
             config.FLEXY['VARIABLES_LOCATION'] = self.template_file
-            config.FLEXY['INSTANCE_NAME_PREFIX'] = self.cluster_name
-            self.merge_flexy_env()
+        config.FLEXY['INSTANCE_NAME_PREFIX'] = self.cluster_name
+        self.merge_flexy_env()
```

Comment:
@dahorak ,
Do we want to merge_flexy_env() even if user has passed the env file ?

For ex consider:
```
1. default_flexy.conf 
FLEXY:
  GIT_PRIVATE_URI: xyz..
  VARIABLES_LOCATION: abc..

2. User supplied env file where he has already updated the above variables:
FLEXY:   
  GIT_PRIVATE_URI: pqr..

Now if we run merge_flexy_env() , 
a. Config parser would have the contents of user supplied env file (which is the value we expect to be passed to flexy run)
b. config.FLEXY would be having contents from default_flexy.conf
c. During merging we iterate over keys of config.FLEXY , in this case we get { GIT_PRIVATE_URI: xyz, VARIABLES_LOCATION: abc}  and config_parser will have  {GIT_PRIVATE_URI: pqr}.
d. We are updating config_parser(key) with value we get in config.FLEXY , which would overwrite what user wanted to pass to flexy, i.e instead of  {GIT_PRIVATE_URI: pqr} , flexy will get { GIT_PRIVATE_URI: xyz}.

am I missing something ?

Reply:
@shylesh My intention and understanding was that any supplied `ocs-ci` configuration should have precedence and user should have under control, what configuration files is providing. There are also variables, which have to be updated anyway (at least `INSTANCE_NAME_PREFIX` with supplied cluster name). The `FLEXY` section in `ocs_ci/framework/conf/default_config.yaml` is empty and I don't see any reference in the code pointing to the `ocs_ci/framework/conf/flexy_default_config.yaml`, am I missing something?

Reply:
@dahorak , Thanks for the clarification.